### PR TITLE
chore: SPM unit tests

### PR DESCRIPTION
**Pull request by jnewc**
# Description

Some changes to CI + testing setup:

* All relevant unit tests moved to SPM test target
* Debug app tests remain in Xcode project unit test target
* Danger step runs in parallel to others
* Pod lint and unit tests no long run on PR edit

# Contributor Checklist

- [x]  All status checks have passed prior to code review
- [ ]  I have added unit tests to a reasonable level of coverage where suitable
- [ ]  I have added UI tests to new user flows, if applicable
- [ ]  I have manually tested newly added UX
- [ ]  I have open a documentation PR, if applicable

# Reviewer Checklist

- [ ]  I have verified that a suitable set of automated tests has been added
- [ ]  I have verified that the title prefix aligns to the code changes + whether a release is expected after merging the PR
- [ ]  I have verified the documentation PR aligns with this PR, if applicable

# Before Merging

- [ ]  If introducing a breaking change, I have communicated it internally
- [ ]  Any related documentation PRs are ready to merge

# Other Stuff

- You can find out more about our automation checks [here](https://primerio.notion.site/iOS-Automation-Checks-198a1eb0e8994d999fb696d5902d97bb)
- Find out more about conventional commits [here](https://primerio.notion.site/Conventional-Commits-6ecfd6a0269a4db2af76d9f0537936b3)

### CODE REVIEW

**Review by borisprimer — APPROVED** [review]
LGTM!

File: .github/workflows/pod_lint.yml [hunk 1]
```diff
@@ -4,7 +4,6 @@ on:
   pull_request:
     types:
       - opened
-      - edited
```
Comment (NQuinn27) [inline .github/workflows/pod_lint.yml:7]: 🙌 Thanks for this + the above. No more rebuilds for edits 😅

File: Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/WebRedirectPaymentMethodTokenizationViewModel.swift [hunk 2]
```diff
@@ -199,14 +199,21 @@ class WebRedirectPaymentMethodTokenizationViewModel: PaymentMethodTokenizationVi
                 )
 
                 Analytics.Service.record(events: [presentEvent, networkEvent])
+
+                guard UIApplication.shared.windows.count > 0 else {
```
Comment (NQuinn27) [inline Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/WebRedirectPaymentMethodTokenizationViewModel.swift:203]: Do we expect this to only fail in tests?

File: Tests/Unit Tests/Mocks.swift [hunk 3]
```diff
@@ -371,7 +371,7 @@ class MockPrimerAPIConfigurationModule: PrimerAPIConfigurationModuleProtocol {
 
     // MARK: - MOCKED PROPERTIES
 
-    var mockedNetworkDelay: TimeInterval = 2
+    var mockedNetworkDelay: TimeInterval = 0.5
```
Comment (NQuinn27) [inline Tests/Unit Tests/Mocks.swift:374]: Nice to see it go in this direction 👌

**Review by NQuinn27 — APPROVED** [review]

**Pull request merged**